lib/Doctrine/ODM/PHPCR/Tools/Helper/UniqueNodeTypeHelper.php

```diff
@@ -0,0 +1,69 @@
+<?php
+/*
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * This software consists of voluntary contributions made by many individuals
+ * and is licensed under the MIT license. For more information, see
+ * <http://www.doctrine-project.org>.
+ */
+
+namespace Doctrine\ODM\PHPCR\Tools\Helper;
+
+use Doctrine\ODM\PHPCR\DocumentManagerInterface;
+use Doctrine\ODM\PHPCR\Mapping\MappingException;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * Provides unique node type mapping verification.
+ */
+class UniqueNodeTypeHelper
+{
+    /**
+     * Check each mapped PHPCR-ODM document for the given document manager,
+     * throwing an exception if any document is set to use a unique node
+     * type but the node type is re-used. If an OutputInterface is provided,
+     * write some basic information to it.
+     *
+     * @param DocumentManagerInterface $documentManager The document manager to check mappings for.
+     * @param OutputInterface $output If provided, output will be written here.
```

Comment:
this is leaking symfony console into the phpcr-odm domain. i would prefer to return an array with debugging information from this method. then the command can do with that whatever it wants. the array could be "FQN" => "information text". the command should then output that if we are in "debug" verbosity